File: cpp/leetcode-maxscoreaftersplittingstring.cpp
#include <string>
#include <algorithm>
using namespace std;

class Solution {
public:
    int maxScore(string s) {
        int maxs = 0;
        int count_zeros = 0;
        int count_ones = count(s.begin(), s.end(), '1');
        for (int i = 0; i < s.length() - 1; i ++) {
            count_zeros += (s[i] == '0');
            count_ones -= (s[i] == '1');
            maxs = max(maxs, count_zeros + count_ones);
        }
        return maxs;
    }
};

File: rs/leetcode-maxscoreaftersplittingstring.rs
impl Solution {
    pub fn max_score(s: String) -> i32 {
        let mut maxs = 0;
        let mut zeros = 0;
        let mut ones = s.chars().filter(|&c| c == '1').count() as i32;

        for (i, c) in s.chars().enumerate().take(s.len() - 1) {
            zeros += if c == '0' { 1 } else { 0 };
            ones -= if c == '1' { 1 } else { 0 };
            maxs = maxs.max(zeros + ones);
        }
        maxs
    }
}